subsys/bluetooth/host/hci_core.c

```diff
@@ -5345,6 +5344,17 @@ int bt_le_adv_start_internal(const struct bt_le_adv_param *param,
 			}
 
 			bt_addr_le_copy(&set_param.direct_addr, peer);
+
+			if (IS_ENABLED(CONFIG_BT_SMP) &&
+			    !IS_ENABLED(CONFIG_BT_PRIVACY) &&
+			    BT_FEAT_LE_PRIVACY(bt_dev.le.features) &&
+			    param->options & BT_LE_ADV_OPT_DIR_ADDR_RPA) {
```

Comment:
put parentheses around (foo & bar) to make it clear that you didn't intend to use &&